watchdog.js

```diff
@@ -0,0 +1,50 @@
+'use strict';
```

Comment:
Done.  Moved `cache.js` and `watchdog.js` into a new `lib/` directory.  In general, I feel like most of the things in the `provisioner/` directory really belong here.  It's confusing that each project's lib directory is not called lib.